packages/react-day-picker/src/contexts/Modifiers/ModifiersContext.tsx

```diff
@@ -0,0 +1,93 @@
+import React from 'react';
+import {
+  isDayPickerMultiple,
+  isDayPickerRange,
+  Modifiers,
+  Matcher
+} from '../../types';
+import { useDayPicker } from '../DayPicker';
+import { useSelectMultiple } from '../SelectMultiple';
+import { useSelectRange } from '../SelectRange';
+
+export const ModifiersContext = React.createContext<Modifiers | undefined>(
+  undefined
+);
+
+export type ModifiersProviderProps = {
+  children: React.ReactNode;
+};
+
+function toMatcherArray(modifierFromProp: Matcher | Matcher[] | undefined) {
+  if (Array.isArray(modifierFromProp)) {
+    return modifierFromProp;
+  } else if (modifierFromProp !== undefined) {
+    return [modifierFromProp];
+  } else {
+    return [];
+  }
+}
+
+/**
+ * Parse the modifiers from the props.
+ *
+ * Internally we want modifiers as an array of matchers – as opposite of the
+ * props which can accept also a matcher.
+ */
+export function ModifiersProvider(props: ModifiersProviderProps): JSX.Element {
+  const context = useDayPicker();
+
+  const {
+    modifiers: customModifiers,
+    selected,
+    disabled,
+    hidden,
+    today,
+    fromDate,
+    toDate
+  } = {
+    ...context
+  };
```

Comment:
Why are you { ...context } here? 🤔

Reply:
Because I have to pass in `context` in full to `isDayPickerMultiple(context)` below, I can't destructure it right away at the useDayPicker() call site.

Reply:
OK but then better write 

```ts
  const {
    modifiers: customModifiers,
    // snip
  } = context;
```

without creating a new object.